packages/ag-charts-enterprise/src/features/annotations/annotationsToolbar.ts

```diff
@@ -23,147 +19,215 @@ interface EventMap {
-
         this.dispatch('pressed-show-menu');
 
-        this.annotationMenu.setAnchor({ x: x + width + 6, y });
+        const index = this.buttons.findIndex((button) => button.value === menu);
+        this.toolbar.toggleActiveButtonByIndex(index);
+        this.annotationMenu.setAnchor({ x: buttonBounds.x + buttonBounds.width + 6, y: buttonBounds.y });
         this.annotationMenu.show<AnnotationType>({
             items,
             ariaLabel: this.ctx.localeManager.t(ariaLabel),
+            class: 'ag-charts-annotations__toolbar-menu',
             sourceEvent: event.sourceEvent,
-            onPress: this.onButtonPressMenuCreateAnnotation.bind(this, event),
+            onPress: this.onButtonPressMenuCreateAnnotation.bind(this, menu),
         });
     }
 
-    private onButtonPressCreateAnnotation(event: _ModuleSupport.ToolbarButtonPressedEvent) {
-        this.ctx.tooltipManager.suppressTooltip('annotations');
-
-        const annotation = stringToAnnotationType(event.value);
-        if (annotation) {
-            this.dispatch('pressed-create-annotation', { annotation });
-        } else {
-            Logger.errorOnce(`Can not create unknown annotation type [${event.value}], ignoring.`);
-        }
-    }
-
     private onButtonPressMenuCreateAnnotation(
-        event: _ModuleSupport.ToolbarButtonPressedEvent<AgToolbarAnnotationsButtonValue>,
+        menu: AnnotationsToolbarButtonValue,
         item: _ModuleSupport.MenuItem<AnnotationType>
     ) {
-        const { toolbarManager } = this.ctx;
-
-        toolbarManager.toggleButton('annotations', event.id, {
-            active: true,
-        });
-        toolbarManager.updateButton('annotations', event.id, {
-            icon: item.icon,
-        });
-
+        const index = this.buttons.findIndex((button) => button.value === menu);
+        this.updateButtonByIndex(index, { icon: item.icon });
         this.dispatch('pressed-create-annotation', { annotation: item.value });
         this.annotationMenu.hide();
     }
 
-    private onCancelled(event: _ModuleSupport.ToolbarCancelledEvent) {
-        if (event.group === 'annotations') {
-            this.dispatch('cancel-create-annotation');
-        }
+    private updateButtonByIndex(index: number, change: Partial<AnnotationsToolbarButtonOptions>) {
+        const button = this.buttons.at(index);
+        if (!button) return;
+
+        button.ariaLabel = change.ariaLabel ?? button.ariaLabel;
+        button.icon = change.icon ?? button.icon;
+        button.label = change.label ?? button.label;
+        button.tooltip = change.tooltip ?? button.tooltip;
+        button.value = change.value ?? button.value;
+
+        this.toolbar.updateButtonByIndex(index, {
+            ariaLabel: button.ariaLabel,
+            icon: button.icon,
+            label: button.label,
+            tooltip: button.tooltip,
+            value: button.value,
+        });
     }
+
+    // TODO: handle esc key
```

Comment:
?

Reply:
There's a couple of key handling things I'm looking into atm, for the final PR on toolbars, including this.